source/opt/eliminate_dead_input_components_pass.cpp

```diff
@@ -60,8 +60,16 @@ Pass::Status EliminateDeadInputComponentsPass::Process() {
         continue;
       }
     }
+    auto stage = context()->GetStage();
     const analysis::Array* arr_type = ptr_type->pointee_type()->AsArray();
     if (arr_type != nullptr) {
+      // Only process array if input of vertex shader, or output of
```

Comment:
Yes, this should be commented better.

Essentially, it is possible to end up with interface incompatibility of arrays between two shaders. This occurs if one shader indexes with a runtime value, and the other only with constant values. This logic was added to prevent such an incompatibility from being created.

I think there are some ways of dealing with this situation, but for now it is beyond the scope of what we were requested to do, so it is simply passed over.

I will add some additional comments to explain this better.